modules/javafx.graphics/src/main/java/com/sun/glass/ui/win/WinTextRangeProvider.java

```diff
@@ -362,6 +362,7 @@ private String GetText(int maxLength) {
         String text = (String)getAttribute(TEXT);
         if (text == null) return null;
         validateRange(text);
+        maxLength = Math.max(-1, maxLength);
```

Comment:
@jperedadnr Thank you so much for the explanation!

> However, the hard part is finding out why it is happening for some Windows users and not for everybody.

Fortunately, there is a reproducer now: Install the DeepL app (https://www.deepl.com/app). I am (unfortunately) serious on that. - https://stackoverflow.com/a/78340343/873282 contains more details for the reproducer.

> why are there reports from users without any such feature (at least that they are aware of)

I would assume that helper apps trying to read from components from the screen use accessibility features to get the text content.

Reply:
I saw your answer, and I've tried it myself, but I'm on those that can't reproduce this issue in any way on my Windows machine (I have latest Windows 11 23H2).

Using DeepL doesn't trigger calls to WinRangeTextProvider for me. 

I get these for instance using Narrator. However, even then, all maxLength values are "normal".

Reply:
It is important to switch on/off the deepl app in the residential area (system tray, lower right corner). In jabref, I can even recover from the crash by closing the deepl app in the system tray, then getting back to it by opening the app again. Pretty funny.

Reply:
@jperedadnr I am collecting all reports at https://github.com/koppor/jfx/pull/2#issuecomment-2061107129. Could you share the output of `ver` (or `winver`) so that we can check if we use the same build?

Reply:
@jperedadnr Because it happened to me while checking the Azure-Windows-Machines: I was admin at these machines. Then, it did not work. I needed to create a user account, allow it to be used using Remote Desktop, connect, login, start both DeepL and JabRef. Then, I could reproduce. - I can share the credentials of the Azure machine in case you have no access Azure or similar cloud services easily.

Reply:
Minor comment: `-1` and `Integer.MAX_VALUE` both seem to be reasonable values for an unlimited length. Maybe, the accessibility developers of Microsoft like the latter more.

The code part `start + maxLength` could be changed to `Math.max(maxLength, start + maxLength)`. Then any value around `Integer.MAX_VALUE` would work.

IMHO `WinTextProvider#GetText` should also be able to handle `Integer.MAX_VALUE-2` - the caller cannot know the values of `start` used, because it does not pass `start` and should not track state of the called instance of `WinTextProvider`.

Reply:
My proposal is in [`5f2a296` (#2)](https://github.com/koppor/jfx/pull/2/commits/5f2a29694c2b5d35210e9383fbcc60588f70f2e1)

```diff
-       int endOffset = maxLength != -1 ? Math.min(end, start + maxLength) : end;
+       int endOffset = maxLength >= 0 ? Math.min(end, Math.max(start + maxLength, maxLength)) : end;
```

Reply:
@jperedadnr The above change leads to a working JabRef. (Thank you for sharing your thoughts with `Integer.MAX_VALUE`.) Now, how to get this into JFX?

Reply:
I just filed https://github.com/openjdk/jfx/pull/1442. If you came up with a better solution, I just withdraw. Just wanted to share it more visible (also (hopefully) to reduce load from you).